ecommerce/views/products.py

```diff
@@ -0,0 +1,129 @@
+    parser_classes = [JSONParser, MultiPartParser, FormParser]
+    filter_backends = [DjangoFilterBackend, filters.SearchFilter]
+    filterset_fields = ["category", "status", "user", "approval_status"]
+    search_fields = ["name", "description"]
+
+    def get_serializer_class(self):
+        """Return appropriate serializer based on action."""
+        if self.request.method == "GET":
+            return ProductListSerializer
+        return CreateProductSerializer
+
+    def get_queryset(self):
+        """
+        Get queryset with appropriate filtering based on user.
+        - Anonymous users see only approved products
+        - Authenticated users can see their own products regardless of approval
+        - Staff can see all products
+        """
+        queryset = super().get_queryset()
+
+        # Optimize queries
+        queryset = queryset.select_related("user", "category").prefetch_related(
+            "images",
+        )
+
+        # Filter based on user permissions
+        if self.request.user.is_authenticated:
+            if self.request.user.is_staff:
+                # Staff can see all products
+                return queryset
+            # Authenticated users see approved products + their own products
+            return queryset.filter(
+                Q(approval_status=Product.APPROVAL_APPROVED)
+                | Q(user=self.request.user),
+            )
+        # Anonymous users only see approved products
+        return queryset.filter(approval_status=Product.APPROVAL_APPROVED)
+
+    def perform_create(self, serializer):
+        """Set the product user to the current user."""
+        serializer.save(user=self.request.user)
+
+
+class ProductDetailView(RetrieveUpdateDestroyAPIView):
+    """
+    View for retrieving, updating, and deleting individual products.
+    GET: Retrieve product details
+    PUT/PATCH: Update product
+    DELETE: Delete product
+    """
+
+    queryset = Product.objects.all()
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+    parser_classes = [JSONParser, MultiPartParser, FormParser]
+    lookup_field = "uuid"
+    lookup_url_kwarg = "pk"
+
+    def get_serializer_class(self):
+        """Return appropriate serializer based on action."""
+        if self.request.method == "GET":
+            return ProductDetailSerializer
+        return UpdateProductSerializer
+
+    def get_queryset(self):
+        """Optimize queries and filter based on user permissions."""
+        queryset = super().get_queryset()
+
+        # Optimize queries
+        queryset = queryset.select_related("user", "category").prefetch_related(
+            "images",
+        )
+
+        # Filter based on user permissions for GET requests
+        if self.request.method == "GET":
+            if self.request.user.is_authenticated:
+                if self.request.user.is_staff:
+                    # Staff can see all products
+                    return queryset
```

Comment:
No need for this condition since the admin should not be using the API endpoint directly.